include/pybind11/cast.h

```diff
@@ -34,6 +34,20 @@ template <typename Type> class opaque {
 
 NAMESPACE_BEGIN(detail)
 
+/// Convert from any type pointer to another avoding warnings about void* vs (void*()) incompatibility.
```

Comment:
Ok, that's a bummer -- seems more like a GCC issue then (inability to disable certain warnings).